community/unsafe/src/main/java/org/neo4j/unsafe/impl/internal/dragons/UnsafeUtil.java

```diff
@@ -350,14 +352,15 @@ public static String newSharedArrayString( char[] chars )
      * The memory is aligned such that it can be used for any data type.
      * The memory is uninitialised, so it may contain random garbage, or it may not.
      */
-    public static long allocateMemory( long sizeInBytes )
+    public static long allocateMemory( long sizeInBytes, MemoryAllocationTracker allocationTracker )
```

Comment:
Since the `MemoryAllocationTracker` is only different in tests, and all other calls refer to the global singleton I would prefer an overload instead.
```java
    public static long allocateMemory( long sizeInBytes )
    {
        return allocateMemory( sizeInBytes, GlobalMemoryTracker.INSTANCE );
    }
```
Same thing could be done for the pimitive collections as well.